Cython/Compiler/ModuleNode.py

```diff
@@ -2841,6 +2851,26 @@ def generate_typeptr_assignment_code(self, entry, code):
                 "%s = &%s;" % (
                     type.typeptr_cname, type.typeobj_cname))
 
+    def generate_pythran_conversion_code(self, env, code):
+        code.putln('''
+template <class T>
+auto to_python_from_expr(T &&value) -> decltype(to_python(
+      typename pythonic::returnable<typename std::remove_cv<typename std::remove_reference<T>::type>::type>::type{std::forward<T>(value)}))
+{
+  using returnable_type = typename pythonic::returnable<typename std::remove_cv<typename std::remove_reference<T>::type>::type>::type;
+  return to_python(returnable_type{std::forward<T>(value)});
+}
+        ''')
+
+    def generate_cpp_inplace_object_destructor(self, env, code):
+        code.putln('''
+template <class T>
+inline static void __cython_destructor(T& o)
+{
+  o.~T();
+}
+	''')
```

Comment:
Fixed in https://github.com/cython/cython/pull/1607/commits/e3fb29c91b13edcc9297b02b0ca10e9a3da3928d